src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -41,16 +41,22 @@ public void setBottomSpeed(double speed) {
     bottomMotor.setSpeed(speed);
   }
 
-  Debouncer speedDebouncer = new Debouncer(0.4);
-  public boolean atSpeed() {
+  Debouncer speedDebouncer = new Debouncer(0.2);
+  public boolean atSpeed(double overrideTopSetpoint, double overrideBottomSetpoint) {
     double tolerance = SPEED_TOLERANCE;
+    overrideTopSetpoint = Math.signum(topMotor.getVelocitySetpoint()) * Math.abs(overrideTopSetpoint);
+    overrideBottomSetpoint = Math.signum(bottomMotor.getVelocitySetpoint()) * Math.abs(overrideBottomSetpoint);
     if(Robot.isSimulation()) {
       tolerance = 3000; // sim setpoint 4k rpm turns out to be like 6661 rpm
     }
-    boolean speedWithinTolerance = ((Math.abs(topMotor.getVelocity() - topMotor.getVelocitySetpoint()) < tolerance) || topMotor.getVelocity() > topMotor.getVelocitySetpoint()) &&
-                             ((Math.abs(bottomMotor.getVelocity() - bottomMotor.getVelocitySetpoint()) < tolerance) || topMotor.getVelocity() > topMotor.getVelocitySetpoint());
+    boolean speedWithinTolerance = ((Math.abs(topMotor.getVelocity() - overrideTopSetpoint) < tolerance) || topMotor.getVelocity() > overrideTopSetpoint) &&
```

Comment:
Redundant code here - create a new method and place boolean calculation in there. ex.

boolean checkSpeedTolerance(tolerance, velocity, setpoint) {
    return (Math.abs(velocity - setpoint) < tolerance) || velocity > setpoint);
}